lib/routes/routeBackbeat.js

```diff
@@ -439,15 +445,22 @@ function putObjectTagging(request, response, log, callback) {
         return callback(err);
     }
     const storageLocation = request.headers['x-scal-storage-class'];
-    const dataStoreVersionId = request.headers['x-scal-data-store-version-id'];
-    const data = [];
-    let totalLength = 0;
-    request.on('data', chunk => {
-        totalLength += chunk.length;
-        data.push(chunk);
-    });
-    request.on('end', () =>
-        parseTagXml(Buffer.concat(data, totalLength), log, (err, tags) => {
+    const sourceVersionId = request.headers['x-scal-source-version-id'];
+    const sourceBucket = request.headers['x-scal-source-bucket'];
+    const sourceTags = request.headers['x-scal-tags'];
+    return metadataGetObject(sourceBucket, request.objectKey,
```

Comment:
did you consider only going to get the updated object metadata if `x-scal-data-store-version-id` is undefined from Backbeat? Or you decided it's better to be consistent and always fetch the updated metadata?

Reply:
I did consider that, but figured it would be simpler to always retrieve it from metadata here. That way, because the value may be `''` in the Kafka entry, we can remove the retrieval of `datastoreVersionId` in backbeat (and sending it as an additional 'x-scal-datastore-version-id' header) and avoid any conditional checks here.

Reply:
Functionally it's equivalent, but there's a non-negligible cost to fetch the metadata on S3 because it may end up being a storage request (unless cached) where we already optimize metadata-only sync to avoid hitting the source S3. I would suggest not to do this request when the source version is readily available in kafka, it should be safe since source version shall be immutable for a particular object version.

Depending on workloads it may not be important or may not even go through this code, but in some kind of workloads it could make a significant difference (e.g. for mass-tagging of objects that already exist and have been synced, this will spare a metadata request for each object).

Code-wise it should be pretty simple with a wrapper of `metadataGetObject` branching to a `process.nextTick(cb, info)` when the version is available locally. It can be an extra test case as well but it's worth it IMO.

Reply:
Okay, I've updated this. I had thought that a request to retrieve Kafka entry values and metadata values would be about the same cost.